presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java

```diff
@@ -336,19 +336,32 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
 
         SchemaTableName schemaTableName = new SchemaTableName(table.getSchemaName(), table.getTableName());
         List<CassandraColumnHandle> columns = cassandraSession.getTable(schemaTableName).getColumns();
-        List<String> columnNames = columns.stream().map(CassandraColumnHandle::getName).collect(Collectors.toList());
-        List<Type> columnTypes = columns.stream().map(CassandraColumnHandle::getType).collect(Collectors.toList());
+        List<String> columnNames = columns.stream()
+                .filter(columnHandle -> !isHiddenIdColumn(columnHandle))
+                .map(CassandraColumnHandle::getName)
+                .collect(Collectors.toList());
+        List<Type> columnTypes = columns.stream()
+                .filter(columnHandle -> !isHiddenIdColumn(columnHandle))
+                .map(CassandraColumnHandle::getType)
+                .collect(Collectors.toList());
+        boolean generateUuid = columns.stream().anyMatch(CassandraMetadata::isHiddenIdColumn);
```

Comment:
Can there be more than one?

Reply:
I suppose it's always one.

Reply:
I guess (and it should), so the code here is too flexible

maybe
```
boolean generateUuid = columns.stream()
  .filter(CassandraMetadata::isHiddenIdColumn))
 .collect(toOptional()) // must be at most one
 .isPresent();
```